docs/core/entity/button.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Button Entity
+sidebar_label: Button
+---
+
+A button entity is an entity that can fire an event / trigger an action towards a device or service but remains stateless from the Home Assistant perspective.
+It can be compared to a real live momentary switch, push-button, or some other form of a stateless switch.
+
+A switch button entity is derived from the  [`homeassistant.components.button.ButtonEntity`](https://github.com/home-assistant/core/blob/dev/homeassistant/components/button/__init__.py),
+and can be helpful for controlling device features like (but not limited to):
+
+- Upgrading firmware
+- Reboot/Restart a device
+- Brew a cup of coffee
+- Reset something (like a counter, filter usage)
+
+If you want to represent something that can be turned on and off (and thus have an actual state), you should use a `switch` entity instead.
```

Comment:
Maybe also mention what to do if you want to integrate a real stateless button in Home Assistant? Ie then you should fire custom events.

Reply:
Done.